### PR TITLE
Added new realtime development task watch:ff.

### PR DESCRIPTION
**Pull request by sun**
Replaces the former `gm` + `watch:gm` tasks to aid in realtime development and debugging.

The new `watch:ff` task builds a new Firefox extension and autoloads it into Firefox in the most performant way possible.

Note that the stated performance goal is not accomplished yet; the current task averages at 2.8 secs (on Windows).

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -193,6 +193,11 @@ module.exports = function(grunt) {
     },
     watch: {
       files: [
+        // Force-ignore artifacts.
+        '!build',
+        '!build/**',
+        '!release',
+        '!release/**',
```
Comment (markhalliwell) [inline Gruntfile.js:206]: Why are these needed? Only the `src` files below are included.

Comment (sun) [inline Gruntfile.js:206]: I really don't know why... I intensively studied all docs + issues for grunt watch, but the watch is continuously interrupted by the build:firefox task.

Comment (markhalliwell) [inline Gruntfile.js:206]: Might it have something to do with the following below this?

``` json
       options: {
          interrupt: true
        }
```

File: Gruntfile.js [hunk 2]
```diff
@@ -271,13 +276,62 @@ module.exports = function(grunt) {
...
+      fallback: -255
+    },
+    function (error, result, code) {
+      if (0 !== code) {
+        grunt.log.errorlns(result.stdout);
+        grunt.log.errorlns(result.stderr);
+      }
+    });
+  });
+  grunt.registerTask('build', ['default', 'compress:chrome', 'build:firefox', 'build:safari']);
```
Comment (sun) [inline Gruntfile.js:335]: These lines are just moved literally/as-is from below.

File: Gruntfile.js (hunk 1 above)
Comment (sun) [inline Gruntfile.js:206]: Nope, the `interrupt` option is fine + works great — successfully avoids to build+autoload an extension in case of two very quickly, consecutively changed/saved files.

Added a clarifying comment for posterity now.
